pandas/io/parsers/python_parser.py

```diff
@@ -893,7 +893,7 @@ def _clear_buffer(self) -> None:
 
     def _get_index_name(
         self, columns: list[Hashable]
-    ) -> tuple[list[Hashable] | None, list[Hashable], list[Hashable]]:
+    ) -> tuple[Sequence[Hashable] | None, list[Hashable], list[Hashable]]:
```

Comment:
Without this, we would get:

pandas/io/parsers/python_parser.py:949: error: Incompatible return value type (got "Tuple[List[Union[Union[str, int, float, bool], Union[Period, Timestamp, Timedelta, Interval[Any]]git , datetime64, timedelta64]], List[Hashable], List[Hashable]]", expected "Tuple[Optional[List[Hashable]], List[Hashable], List[Hashable]]")  [return-value]